gadget/validate.go

```diff
@@ -77,6 +77,10 @@ type ValidationConstraints struct {
 // are handled identically to ReadInfo(). Optionally takes additional validation
 // constraints, which for instance may only be known at run time,
 func Validate(gadgetSnapRootDir string, model Model, extra *ValidationConstraints) error {
+	// TODO: also validate that only one "<bl-name>.conf" file is in the root
```

Comment:
On top of that, I think we should validate that the bootloader in gadget.yaml has a conf file inside. Ideally ForGadget could use that knowledge too.